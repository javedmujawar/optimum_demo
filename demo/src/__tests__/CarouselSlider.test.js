import { render, screen } from "@testing-library/react";
import CarouselSlider from "../components/Carousel/CarouselSlider";

describe("CarouselSlider Component", () => {
  const mockSteps = [
    { title: "Step 1" },
    { title: "Step 2" },
    { title: "Step 3" },
  ];

//   test("renders all steps", () => {
//     render(<CarouselSlider steps={mockSteps} currentStep={0} onSelectOption={() => {}} />);
//     const slides = screen.getAllByRole("region");
//     expect(slides).toHaveLength(mockSteps.length);
//   });

//   test("applies the active class to the current step", () => {
//     const currentStep = 1;
//     render(<CarouselSlider steps={mockSteps} currentStep={currentStep} onSelectOption={() => {}} />);
//     const activeSlide = screen.getByRole("region", { name: /step 2/i });
//     expect(activeSlide).toHaveClass("active");
//   });

//   test("does not apply the active class to non-current steps", () => {
//     const currentStep = 1;
//     render(<CarouselSlider steps={mockSteps} currentStep={currentStep} onSelectOption={() => {}} />);
//     const inactiveSlides = screen.getAllByRole("region").filter((slide) => !slide.classList.contains("active"));
//     expect(inactiveSlides).toHaveLength(mockSteps.length - 1);
//   });

  test("renders correctly with an empty steps array", () => {
    render(<CarouselSlider steps={[]} currentStep={0} onSelectOption={() => {}} />);
    const slides = screen.queryAllByRole("region");
    expect(slides).toHaveLength(0);
  });

  test("matches the snapshot", () => {
    const { container } = render(<CarouselSlider steps={mockSteps} currentStep={0} onSelectOption={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
