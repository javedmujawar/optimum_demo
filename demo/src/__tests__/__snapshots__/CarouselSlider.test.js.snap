// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlider Component matches the snapshot 1`] = `
<div>
  <div
    class="carouselSliderMain"
  >
    <div
      class="carousel-slide active"
    >
      <div
        class="options"
      />
    </div>
    <div
      class="carousel-slide "
    >
      <div
        class="options"
      />
    </div>
    <div
      class="carousel-slide "
    >
      <div
        class="options"
      />
    </div>
  </div>
</div>
`;
